src/app/aether-application/application/application-datasource.ts

```diff
@@ -13,106 +12,90 @@ import {
-            ['enterprise-id', 'application-id']
+            enterpriseService,
+            undefined,
+            ['application-2.1.0'],
+            ['application-id']
         );
     }
 
     loadData(
-        dataSourceObservable: Observable<Enterprises>,
+        dataSourceObservable: Observable<ApplicationList>,
         onDataLoaded: (
-            dataSourceThisScope: RocDataSource<
-                EnterprisesEnterpriseApplication,
-                Enterprises
-            >
-        ) => void
+            dataSourceThisScope: RocDataSource<Application, ApplicationList>
+        ) => void,
+        enterpriseId?: TargetName
     ): void {
-        dataSourceObservable
-            .pipe(
-                map((x: Enterprises) => x?.enterprise),
-                skipWhile((x) => x === undefined),
-                mergeMap((items: EnterprisesEnterprise[]) => from(items))
-            )
-            .subscribe(
-                (value: EnterprisesEnterprise) => {
-                    if (value.application) {
-                        value.application.forEach((app) => {
-                            app['enterprise-id'] = value['enterprise-id'];
-                            const fullPath = this.deletePath(
-                                value['enterprise-id'],
-                                app['application-id']
-                            );
-                            if (this.bs.containsDeleteEntry(fullPath)) {
-                                app[FORDELETE] = STRIKETHROUGH;
-                            }
-                            // Check for usages in slices
-                            if (value.site) {
-                                value.site.forEach((site) => {
-                                    if (site.slice) {
-                                        site.slice.forEach((slice) => {
-                                            if (slice.filter) {
-                                                slice.filter.forEach(
-                                                    (filter) => {
-                                                        if (
-                                                            filter.application ===
-                                                            app[
-                                                                'application-id'
-                                                            ]
-                                                        ) {
-                                                            app[ISINUSE] =
-                                                                'true'; // Any match will set it
-                                                        }
-                                                    }
-                                                );
+        dataSourceObservable.pipe(skipWhile((x) => x === undefined)).subscribe(
+            (appList: ApplicationList) => {
+                appList.forEach((app: Application) => {
```

Comment:
This is most likely for a different patch, but should we look into generalizing this method to highlight the deleted items as we did for `merge` method to combine data in the basket with data from the APIs?

Reply:
yeah a different patch - this one is too big already